deepmd/utils/plugin.py

```diff
@@ -93,3 +96,60 @@ class PluginVariant(metaclass=VariantABCMeta):
     """A class to remove `type` from input arguments."""
 
     pass
+
+
+def make_plugin_registry(name: Optional[str] = None) -> Type[object]:
```

Comment:
I also added a `make_plugin_registry` method and will apply it to other classes after this PR.